pip/req/req_install.py

```diff
@@ -421,6 +421,14 @@ def run_egg_info(self):
                     self.pkg_info()["Version"],
                 ]))
             self._correct_build_location()
+        else:
+            metadata_name = canonicalize_name(self.pkg_info()["Name"])
+            if canonicalize_name(self.req.project_name) != metadata_name:
+                raise InstallationError(
+                    'Running setup.py (path:%s) egg_info for package %s '
```

Comment:
pip errors just after cloning the repository, so the error seemed sufficient to me:

    Obtaining djpyfs from git+https://github.com/pmitros/django-pyfs.git#egg=djpyfs (from -r /home/xfernandez/some_requirements.txt (line 7))
      Cloning https://github.com/pmitros/django-pyfs.git to ./src/djpyfs
    Running setup.py (path:/home/xfernandez/.virtualenvs/d694c7c01647e1aa/src/djpyfs/setup.py) egg_info for package djpyfs produced metadata for project name django-pyfs